### PR TITLE
fix: `read_gbq_table` respects primary keys even when `filters` are set

### PR DESCRIPTION
**Pull request by tswast**
Closes internal issues 338039517 (primary key inconsistency), 338037499 (LIMIT for max_results), 340540991 (avoid running query immediately if time travel is supported), 337925142 (push down column filters to when we create the time travel subquery).

feat: `read_gbq_query` supports `filters`
perf: use a `LIMIT` clause when `max_results` is set
perf: don't run query immediately from `read_gbq_table` if `filters` is set

🦕

### CODE REVIEW

**Review by TrevorBergeron — COMMENTED** [review]
mostly looks good, seems we've lost some validation on unknown column names though?

File: bigframes/session/__init__.py [hunk 1]
```diff
@@ -690,16 +686,67 @@ def _read_gbq_table(
...
-            time_travel_timestamp,
+            ibis_client=self.ibis_client,
+            table_ref=table_ref,
+            index_cols=index_cols,
+            columns=columns,
+            filters=filters,
+            time_travel_timestamp=time_travel_timestamp,
         )
 
         for key in columns:
```
Comment (tswast) [inline bigframes/session/__init__.py:752]: @TrevorBergeron Which "validation for unknown column names'" are we missing? I still see this here.

Comment (tswast) [reply bigframes/session/__init__.py:752]: Would you like me to move it to before we create the ibis expression? It's possible ibis will fail with a dry run error when we try to access the columns in it now.

Comment (tswast) [reply bigframes/session/__init__.py:752]: I see `tests/system/small/test_session.py::test_read_gbq_w_columns[unknown_col]` failure now. Likely this is the issue. I'll move this validation to after we tech the table metadata.

Comment (tswast) [reply bigframes/session/__init__.py]: Fixed in 87365db9d494cdc6cfd18ba930dea2b788c591de

**Review by TrevorBergeron — APPROVED** [review]

**Pull request merged**